pycontrails/models/sac.py

```diff
@@ -142,24 +148,15 @@ def eval(
         self.source["rh_critical_sac"] = rh_crit_sac
         self.source["sac"] = sac_
 
-        # Tag output with additional attrs when source is MetDataset
-        if isinstance(self.source, MetDataset):
-            attrs: dict[str, Any] = {
-                "description": self.long_name,
-                "pycontrails_version": pycontrails.__version__,
-            }
-            if scale_humidity:
-                for k, v in self.params["humidity_scaling"].description.items():
-                    attrs[f"humidity_scaling_{k}"] = v
-            if self.met is not None:
-                attrs["met_source"] = self.met.attrs.get("met_source", "unknown")
-            if isinstance(engine_efficiency, (int, float)):
-                attrs["engine_efficiency"] = engine_efficiency
-
-            self.source["sac"].data.attrs.update(attrs)
-            return self.source["sac"]
-
-        # In GeoVectorDataset case, return source
+        # Tag output with additional metadata attrs
+        self.transfer_met_source_attrs()
+        self.source.attrs["pycontrails_version"] = pycontrails.__version__
+        if scale_humidity:
+            for k, v in humidity_scaling.description.items():
+                self.source.attrs[f"humidity_scaling_{k}"] = v
+        if isinstance(engine_efficiency, (int, float)):
+            self.source.attrs["engine_efficiency"] = engine_efficiency
```

Comment:
should we put fuel on output since we're not copying it from source anywhere?

Reply:
The main reason why I didn't is because our `Fuel` instance isn't harmonious with `self.source.data.to_netcdf(...)`. We'd have to handle it some other way ... like with how we handle `humidity_scaling` just above this for example.

There are a few other similar examples of this sort of thing kicking around in pycontrails. We have certain metadata that we keep in a dictionary (`Models.params` or `MetDataset.attrs` or `Flight.attrs`), and we want these to be serializable to JSON or netCDF or parquet. To do this right is probably a fairly invasive change.

Reply:
Ah ok, good point.